website/docs/rbs-support.md

```diff
@@ -699,6 +699,93 @@ class Foo
 end
 ```
 
+## Generic classes and modules
+
+RBS supports generic classes and modules.
+
+[Type members](generics#type_member--type_template) can be specified using a
+`#:` comment on the class or module:
+
+```ruby
+#: [E]
+class Box
+  #: -> void
+  def initialize
+    @elems = [] #: T::Array[E]
+  end
+
+  #: (E) -> void
+  def <<(e)
+    @elems << e
+  end
+end
+
+box = Box.new #: Box[Integer]
+box << 42
+```
+
+RBS generics do not use `T::Generic`, thus the `[]` method doesn't exist at
+runtime and Sorbet will report an error if you try to use it:
+
+```ruby
+Box[Integer].new
+   ^^^^^^^^^ error: Method `[]` does not exist on `T.class_of(Box)`
+```
+
+To define a type template, use the `#:` comment on a `class << self`:
```

Comment:
Note that defining a `type_member` in a singleton class is not the same as defining a `type_template` on the attached class:

https://sorbet.run/#%23%20typed%3A%20true%0A%0Aclass%20Module%3B%20include%20T%3A%3ASig%2C%20T%3A%3AGeneric%3B%20end%0A%0Aclass%20ExampleGood%0A%20%20X%20%3D%20type_template%0A%0A%20%20sig%20%7B%20returns%28X%29%20%7D%0A%20%20def%20self.x1%20%3D%20raise%0A%0A%20%20class%20%3C%3C%20self%0A%20%20%20%20sig%20%7B%20returns%28X%29%20%7D%0A%20%20%20%20def%20x2%20%3D%20raise%0A%20%20end%0Aend%0A%0Aclass%20ExampleBad%0A%20%20class%20%3C%3C%20self%0A%20%20%20%20X%20%3D%20type_member%0A%20%20%20%20sig%20%7B%20returns%28X%29%20%7D%0A%20%20%20%20def%20x2%20%3D%20raise%0A%20%20end%0A%0A%20%20sig%20%7B%20returns%28X%29%20%7D%0A%20%20def%20self.x1%20%3D%20raise%0Aend

Reply:
Can't you get around by moving `x1` into `class << self`?